### ISSUE TITLE
LongName generating when ShortName provided

### ISSUE THREAD

**Issue by progala2** [bilal-fazlani/commanddotnet#183]
Hi,

It is possible to enforce adding the LongName property automatically when the ShortName one is provided?
Because it is annoying when I have the kebab case formatting set up and I add the ShortName to the option and the LongName is not generated anymore 😢 
I'm using the 3.0 version.

PS As it is my first issue here, I want to thank you for great work! 😄

**Comment by drewburlingame** [comment]
@progala2 this is [fixed in v3](https://github.com/bilal-fazlani/commanddotnet/blob/beta-v3/master/CommandDotNet.Tests/CommandDotNet.NameCasing/CaseTests.cs#L84).  V3 is feature complete, I just need to finish [the documentation](https://bilal-fazlani.github.io/commanddotnet/V3/v3-changes/).

**Comment by drewburlingame** [comment]
I misread this issue in my previous response.  This is an issue I find annoying as well and because the feature already exists, I'm not sure how best to change the behavior without breaking existing applications.  What I've come up with so far is to create an AppSetting that will keep the long names when a short name is provided.  When enabled, then this behaves as you've described, but it also means you can't have options with only a short name.  We could then add OptionAttribute.NoLongName, but this only makes sense when the other option is set.  It's not required otherwise and thus confusing.  It's the better option if keeping the long name is what most devs would prefer.  It would be great to get some feedback from the community on this.  I'm open to other options if you have any.

With v3, kebab casing can be applying to long name overrides as shown [in these tests](https://github.com/bilal-fazlani/commanddotnet/blob/beta-v3/master/CommandDotNet.Tests/CommandDotNet.NameCasing/CaseTests.cs#L84) so you could provide the long name as Pascal case and it will be converted to kebab.  Best practice will be to use the camel or pascal case version of the name as those can be converted to any other case.  Humanizer is not able to convert kebabcase to other cases.

**Comment by drewburlingame** [comment]
ps, what do you think of the 3.0 version?  I'd appreciate any feedback you have, positive or otherwise.

**Comment by progala2** [comment]
- I really like the split between different nugets, so there is no one huge dll.
- The new naming of attributes also is a good thing.
- DI BY CONSTRUCTOR ❤️ 😂 (this is the main reason I've changed to v3.0)
- Middleware concept
- Test tools

From the things to improve:
- I hope the issue I post here will be solved somehow 😬 
- Some stuff(methods) hasn't been yet documented. I wanted to do something with parsing Operands/Options and it wasn't clear from the method arguments what to do. But I think it is the right of the RC version.
It is good when methods are self-explanatory (or there is some comment if it is hard to do)

**Comment by drewburlingame** [comment]
thanks for the feedback @progala2.  I'm glad you're liking the the updates.  What were you trying to do with Operands/Options that wasn't clear so I can be sure to include documentation for it.  I tried to make the names more self-explanatory where I could, but I'm sure there are places I missed.  Please let me know if you have ideas for improvements.  After the scope of work I took on here, it's easy for stuff like that to fall through the cracks.

**Comment by drewburlingame** [comment]
@bilal-fazlani I've been thinking about this one for a while.  At the very least, I think we need an AppSetting that makes ShortName additive so that we can add a ShortName but don't need to explicitly add a LongName.   When we have that, we still need to support cases where only the ShortName should be used.  I think in those cases LongName could be set to "" or null with something like
```c#

        private string _longName;

        public string LongName
        {
            get => _longName;
            set
            {
                IgnoreLongName = value.IsNullOrEmpty();
                _longName = value;
            }
        }

        public bool IgnoreLongName { get; private set; }
``` 
I think more often, devs are looking just to add a short name, not replace the long name.  This will make that use case simpler.

What do you think?

**Comment by bilal-fazlani** [comment]
> I think more often, devs are looking just to add a short name, not replace the long name. 
I agree.

If that is the case then we should not only support it but also make it default behaviour.

So what should user code look like when s/he just wants to add short name and what should it look like when they just wants to have a short name?

**Comment by drewburlingame** [comment]
If we make it the default behavior, it could be a breaking change.  Existing apps could suddenly have long names when they only wanted short names.  We could start with the settings defaulting to off and then introduce the breaking change with the next major version update.

**Comment by bilal-fazlani** [comment]
Yes 👍

**Comment by drewburlingame** [comment]
@progala2 I've implemented this feature in v 3.5.0.  You can read about it here https://commanddotnet.bilal-fazlani.com/DefiningCommands/arguments/#shortname-and-longname-interactions
